pennylane/pulse/transmon.py

```diff
@@ -215,3 +217,263 @@ def __add__(self, other):
             return TransmonSettings(new_connections, new_omega, new_g, anharmonicity=new_anh)
 
         return self
+
+
+def transmon_drive(amplitude, phase, freq, wires, d=2):
+    r"""Returns a :class:`ParametrizedHamiltonian` representing the drive term of a transmon qubit.
+
+    The Hamiltonian is given by
+
+    .. math::
+
+        \Omega(t) \left(e^{i (\phi(t) + \nu t)} a_q + e^{-i (\phi(t) + \nu t)} a^\dagger_q\right)
+
+    where :math:`[a^\dagger_p, a_q] = i \delta_{pq}` are bosonic creation and annihilation operators
+    and :math:`q` is the qubit label (``wires``).
+    The arguments ``amplitude``, ``phase`` and ``freq`` correspond to :math:`\Omega`, :math:`\phi`
+    and :math:`\nu`, respectively, and can all be either fixed numbers (``float``) or depend on time
+    (``callable``). If they are time-dependent, they need to abide by the restrictions imposed
+    in :class:`ParametrizedHamiltonian` and have a signature of two parameters, ``(params, t)``.
+
+    For realistic simulations, one may restrict the amplitude, phase and drive frequency parameters.
+    For example, the authors in `2008.04302 <https://arxiv.org/abs/2008.04302>`_ impose the restrictions of
+    a maximum amplitude :math:`\Omega_{\text{max}} = 20 \text{MHz}` and the carrier frequency to deviate at most
+    :math:`\nu - \omega = \pm 1 \text{GHz}` from the qubit frequency :math:`\omega`
+    (see :func:`~.transmon_interaction`).
+    The phase :math:`\phi(t)` is typically a slowly changing function of time compared to :math:`\Omega(t)`.
+
+    .. note:: Currently only supports ``d=2`` with qudit support planned in the future. For ``d=2`` we have :math:`a:=\frac{1}{2}(\sigma^x + i \sigma^y)`.
+
+    .. note:: Due to convention in the respective fields, we ommit the factor :math:`\frac{1}{2}` present in the related constructors :func:`~.drive` and :func:`~.rydberg_drive`
+
+    .. seealso::
+
+        :func:`~.drive`, :func:`~.rydberg_drive`, :func:`~.transmon_interaction`
+
+    Args:
+        amplitude (Union[float, callable]): The amplitude :math:`\Omega(t)`.
```

Comment:
In `transmon_interaction` we specify that frequency is in GHz. We should consider having similar indication of expected or recommended units here.

Reply:
done :) ✅ 

Feels kind of weird to put this since in this function nothing is actually happening in terms of units, but will be important to bridge with real hardware